### PR TITLE
bugfix: ZENKO-1365 Orbit set https for RING+S3C

### PR DESCRIPTION
**Pull request by dora-korpar**
Add HTTPS support for RING+S3C endpoints added through Orbit.

### CODE REVIEW

**Comment by dora-korpar** [comment]
@bert-e reset

File: lib/management/configuration.js [hunk 1]
```diff
@@ -116,6 +116,19 @@ function patchConfiguration(newConf, log, cb) {
                     break;
                 case 'location-scality-ring-s3-v1':
                     pathStyle = true; // fallthrough
```
Comment (alexanderchan-scality) [inline lib/management/configuration.js]: no longer a `fallthrough`

**Review by rachedbenmustapha — COMMENTED** [review]
This is a lot of copy-pasted code that was not needed before, can we track down the regression and fix that instead?

File: lib/management/configuration.js [hunk 2]
```diff
@@ -116,6 +116,19 @@ function patchConfiguration(newConf, log, cb) {
...
+                    let https = false;
+                    let ringEndpoint = l.details.endpoint ||
+                        's3.amazonaws.com';
+                    if (ringEndpoint.includes('://')) {
+                        const url = new URL(awsEndpoint);
+                        ringEndpoint = url.host;
+                        https = url.scheme === 'https';
+                    }
+                    location.details = {
+                        awsEndpoint: ringEndpoint,
```
Comment (alexanderchan-scality) [inline lib/management/configuration.js]: missing properties? `credentials`, `bucketName`, `bucketMatch`, sse, `supportsVersioning`, `pathStyle`, `https`.

Comment (dora-korpar) [reply lib/management/configuration.js]: Are these set by default somewhere if nothing is specified for location.details? Cos what is being set doesn't seem consistent across the aws-derived locations

Comment (rachedbenmustapha) [reply lib/management/configuration.js]: They were set in the fallthrough code path

Comment (alexanderchan-scality) [reply lib/management/configuration.js]: the defaults are empty.

**Comment by dora-korpar** [comment]
@bert-e reset

**Comment by dora-korpar** [comment]
It turned out the url object was being used incorrectly

**Comment by dora-korpar** [comment]
@bert-e reset

**Review by philipyoo — APPROVED** [review]

**Review by rachedbenmustapha — APPROVED** [review]

**Comment by dora-korpar** [comment]
@bert-e approve

**Review by alexanderchan-scality — APPROVED** [review]

**Pull request merged**